plutus-book/doc/09-extended.adoc

```diff
@@ -62,83 +66,133 @@ sections of a Haskell program, called Plutus Tx, which are then compiled into
+For a draft of the formal specification of the extended UTXO model with
 scripts, see <<endmatter#scripts>>.
 
-Scripts, like witnesses, are another
-tool to make sure a transaction is performing only authorized actions.
-The purpose of witnessing for (non-script) spending of outputs is to verify
+The purpose of witnessing is to verify
 that the wallet spending the output has the private key associated with the address
-in the corresponding UTxO entry (i.e. the address from which it is spending the
+in the corresponding UTXO entry (i.e. the address from which it is spending the
 money) by checking that it was indeed signed (witnessed)
 by that key.
+Scripts, like witnesses, are another
+tool to ensure a transaction is spending only outputs it is authorized to.
+
+A script to which it
+is possible to pay has its own address in the UTXO, and it is the hash of this script.
+Only spending from these _script addresses_ can be controlled by the script (whose
+address this is). Let us look at a very small example that omits some of the
+details, but gives the reader an idea of the kind of thing scripts can do.
+Suppose the owner of (non-script) address `a` has promised to pay `x` Ada
+to the first person to guess a word they are thinking of.
+This is a kind of contract, and we can interpret it as a smart contract by
+considering exactly how it controls the transfer of funds. This process
+always starts with a contribution of funds to the contract.
+
+Recall that in the classic model, UTXO entries are of the form
+`(txid, ix) |-> (a,c)`. The pair `(a,c)` has type `TxOut` and is called an output,
+where `a` is the address of the owner of the funds
+(the paying or staking public key, usually), and `c` is the value of Ada in the output.
+
+Roughly, we expect the owner of `a` to pay to the contract `c` Ada, which can
+only be spent if the correct word guess is provided by the participant.
+We also notice here that the secret word cannot be stored in plain text on
+the blockchain, so we must only ever expose the hash of this word.
+So, the contract we want should take a hashed secret word and a plain text
+word guess provided by a participant, and allow him to spend the Ada `c`
+locked at the contract address if the guess is correct.
+To express this, we write in `PlutusTx` the following contract,
+representing the condition under which the transfer of funds can take place:
+
+----
+s = "given the hash hs of a secret word, and a word guess wg,
+check that wg hashes to hs"
+----
+
+The contract is created when the owner of `a` pays to this script (i.e. to the
+address `hash s`) `c` Ada. So, he produces a transaction that contains some inputs spending
+UTXO entries with `a` in the address field, and an output
+
+----
+ix |-> (hash s, c, hs)
+----
+
+Note that the hash of the secret word `hs` is a separate piece of data in the
+output - we will explain
+why in more detail when we give the validator-redeemer-data script model overview.
```

Comment:
I recommend doing an example where the hash is baked into the validator script at first. This allows you to explain validators and redeemers without needing data scripts.